src/Utils/obiDialogParser.ts

```diff
@@ -203,14 +203,44 @@ export class ObiDialogParser {
         steps: (string | OBITypes.OBIDialog)[],
         conditionalEntities: { [key: string]: Set<string> },
         requiredCondition?: OBIUtils.ConditionEntityAndValue): Promise<void> {
-        for (const step of steps) {
+        for (const [i, step] of steps.entries()) {
             if (typeof step === "string") {
                 this.warnings.push(`Unexpected string step in ${node.dialog.$id}`)
                 continue
             }
             // Handle any steps that may contain an expansion of the dialog tree.
             switch (step.$type) {
-                case OBIStepType.BEGIN_DIALOG:
+                case OBIStepType.HTTP_REQUEST: {
+                    // For HTTP requests, we need to inject a synthetic utterance in to the dialog tree to represent a user
+                    // supplying the input to the HTTP request.
+                    // To do this, we split apart both the dialog tree and the steps within the OBI dialog.
+                    // The HTTP request goes in a new node with a GUID representing the user utterance.
+
+                    // Build a node containing just the HTTP request.
+                    // TODO(thpar) : consider handling the case where the HTTP request is in a SwitchCondition branch (this may be unusual).
+                    // In the SwitchCondition case, the step is in the switch's branch, not the node's dialog.
+                    const httpStep = node.dialog.steps![i]
```

Comment:
Re : scorer steps appended to previous TrainRound - that's already implemented, it happens here : https://github.com/microsoft/ConversationLearner-UI/blob/0f14ab35d1f0c075c0b1e383745680be2a86e90b/src/Utils/obiDialogParser.ts#L373

Re : Why do I add 2 nodes, rather than making a single cut before the HTTP request ?
That's in this case : https://github.com/microsoft/ConversationLearner-UI/blob/0f14ab35d1f0c075c0b1e383745680be2a86e90b/src/Utils/obiDialogParser.ts#L213

There's one node generated for the HTTP request action and one for the steps following the HTTP action.

> What's the difference between the steps passed to the function and the node.dialog.steps?
There are 2 code paths that invoke this function.  The first is from the `steps` in a OBI Dialog's top level -- this is node.dialog.steps -- and the second is from `steps` within a `SwitchCondition` branch.
![steps](https://user-images.githubusercontent.com/53840199/68510924-9d0cd500-0229-11ea-9ec4-35c9d13b7acb.png)

I'm not sure that we'll actually encounter an HTTP call in case #2, so I'm punting on it for now w/ a TODO.  Handling it correctly would add a fair amount of complexity (or a large refactor) that I'd rather not incur until we need to.